src/map/vending.hpp

```diff
@@ -30,5 +30,6 @@ void vending_vendinglistreq(struct map_session_data* sd, int id);
 void vending_purchasereq(struct map_session_data* sd, int aid, int uid, const uint8* data, int count);
 bool vending_search(struct map_session_data* sd, t_itemid nameid);
 bool vending_searchall(struct map_session_data* sd, const struct s_search_store_search* s);
+void vending_update_vendor_location(map_session_data &sd);
```

Comment:
```suggestion
void vending_update(map_session_data &sd);
```
After your last commit it is not only location that is getting updated.